src/oss-tests/ProjectManagerTests/NuGetProjectManagerTests.cs

```diff
@@ -23,37 +27,92 @@ public class NuGetProjectManagerTests
-        public async Task MetadataSucceeds(string purlString, string? description = null)
+        [DataRow("pkg:nuget/razorengine@4.2.3-beta1", "RazorEngine - A Templating Engine based on the Razor parser.", "Matthew Abbott, Ben Dornis, Matthias Dittrich")] // Normal package
+        [DataRow("pkg:nuget/razorengine", "RazorEngine - A Templating Engine based on the Razor parser.", "Matthew Abbott, Ben Dornis, Matthias Dittrich", "4.5.1-alpha001")] // Normal package, no specified version
+        public async Task MetadataSucceeds(string purlString, string? description = null, string? authors = null, string? latestVersion = null)
         {
             PackageURL purl = new(purlString);
+            IManagerPackageActions<NuGetPackageVersionMetadata>? nugetPackageActions = PackageActionsHelper<NuGetPackageVersionMetadata>.SetupPackageActions(
+                purl,
+                JsonConvert.DeserializeObject<NuGetPackageVersionMetadata>(_metadata[purl.ToString()]),
+                JsonConvert.DeserializeObject<IEnumerable<string>>(_versions[purl.ToString()]));
+            _projectManager = new NuGetProjectManager(".", nugetPackageActions, _mockFactory.Object);
+
             PackageMetadata metadata = await _projectManager.GetPackageMetadata(purl, useCache: false);
 
             Assert.AreEqual(purl.Name, metadata.Name, ignoreCase: true);
-            Assert.AreEqual(purl.Version, metadata.PackageVersion);
+            
+            // If a version was specified, assert the response is for this version, otherwise assert for the latest version.
+            Assert.AreEqual(!string.IsNullOrWhiteSpace(purl.Version) ? purl.Version : latestVersion,
+                metadata.PackageVersion);
             Assert.AreEqual(description, metadata.Description);
+            if (!string.IsNullOrWhiteSpace(authors))
+            {
+                List<User> authorsList = authors.Split(", ").Select(author => new User() { Name = author }).ToList();
+                CollectionAssert.AreEquivalent(authorsList, metadata.Authors);
+            }
         }
         
+        [DataTestMethod]
+        [DataRow("pkg:nuget/razorengine@4.2.3-beta1", 84, "4.5.1-alpha001")]
+        [DataRow("pkg:nuget/razorengine", 84, "4.5.1-alpha001")]
+        [DataRow("pkg:nuget/razorengine", 40, "3.10.0", false)]
+        public async Task EnumerateVersionsSucceeds(
+            string purlString, 
+            int count, 
+            string latestVersion, 
+            bool includePrerelease = true)
+        {
+            PackageURL purl = new(purlString);
+            IManagerPackageActions<NuGetPackageVersionMetadata>? nugetPackageActions = PackageActionsHelper<NuGetPackageVersionMetadata>.SetupPackageActions(
+                purl,
+                JsonConvert.DeserializeObject<NuGetPackageVersionMetadata>(_metadata[purl.ToString()]),
+                JsonConvert.DeserializeObject<IEnumerable<string>>(_versions[purl.ToString()])?.Reverse(),
```

Comment:
Nit: would you mind elaborating on why the versions list is reversed here? Is the test data actually sorted in latest-first order, and we have to reverse it to properly emulate oldest-first order returned by the actual API/SDK?

Reply:
The test data is sorted in ascending order, and as discussed in https://github.com/microsoft/OSSGadget/pull/327#discussion_r849883922, I added to the documentation that the `PackageActionsHelper` expects the list of versions in descending order, so I reversed it here. I suppose I could have it reverse in the helper itself and keep it out of it here.
Thoughts?

Reply:
Got it, that makes sense. I don't think you need to change anything - just trying to keep track of all of the order reversals involved here 😅